opr/assetList.go

```diff
@@ -38,11 +61,24 @@ func (o OraclePriceRecordAssetList) List(version uint8) []Token {
 	tokens := make([]Token, len(assets))
 	for i, asset := range assets {
 		tokens[i].Code = asset
-		tokens[i].Value = o[asset]
+		tokens[i].Value = o.Value(asset)
```

Comment:
And here you are again converting your uint64 to float64. So I don't get the motivation to use uint64 as you are literally always converting it to float64

Reply:
If you are not using uint64, then I suggest to stick with float64, and later have a V3 that is actually using uint64 everywhere. I don't see the benefit of doing it halfway

Reply:
> If you are not using uint64, then I suggest to stick with float64, and later have a V3 that is actually using uint64 everywhere. I don't see the benefit of doing it halfway

the deciding factor here is actually the format we get from the APIs. there's no math involved in writing OPRs to the chain, it's just a 1:1 transfer from the API to the record. 

our apis are providing the exchange rate as float64 as far as i can tell (i checked 1forge, cmc, currencylayer), so it makes sense to keep them as float64 in the opr itself.

it's only when we start doing math (both addition and multiplication) with them that it matters we use some sort of fixed point or whole number system in order to improve accuracy. 

so i suppose the question is: do we need improved accuracy during the grading process? i don't think so, since the results of the grade are not used for anything so any float error doesn't compound or has any impact 

converting them to a fixed format in the opr saves us some processing power later but that's probably negligible

Reply:
@WhoSoup Explained it perfectly. I still think it'd be beneficial to move away from floats for the grading too, but at this time, I don't see a huge pressure too. With conversions on the docket, I want to ensure the rates put on the chain are the same rates pulled and used.